### ISSUE TITLE
Model detects object with confidence score less than the specified threshold

### ISSUE THREAD

**Issue by KuldeepDileep** [Deci-AI/super-gradients#1841]
### 💡 Your Question

I am using YOLO-NAS for object detection. Although I have set the _score_thres_ to 0.8, when I test the model it also detects object with confidence score less than 0.8. I've included a frame from the video below for reference, where it can be seen that the confidence scores of both predicted anchor boxes are below 0.8.

Code: 

`from super_gradients.training import models
model = models.get('yolo_nas_l', 
                   num_classes=len(dataset_params['classes']), 
                   pretrained_weights="coco"
                   )
model = model.to("cuda" if torch.cuda.is_available() else 'cpu')`

`train_params = {
    # ENABLING SILENT MODE
    'silent_mode': False,
    "average_best_models":True,
    "warmup_mode": "linear_epoch_step",
    "warmup_initial_lr": 1e-6,
    "lr_warmup_epochs": 3,
    "initial_lr": 5e-4,
    "lr_mode": "cosine",
    "cosine_final_lr_ratio": 0.1,
    "optimizer": "Adam",
    "optimizer_params": {"weight_decay": 0.0001},
    "zero_weight_decay_on_bias_and_bn": True,
    "ema": True,
    "ema_params": {"decay": 0.9, "decay_type": "threshold"},
    "mixed_precision": True,
    "loss": PPYoloELoss(
        use_static_assigner=False,
        # NOTE: num_classes needs to be defined here
        num_classes=len(dataset_params['classes']),
        reg_max=16
    ),
    "valid_metrics_list": [
        DetectionMetrics_050_095(
            score_thres=0.8,
            top_k_predictions=300,
            # NOTE: num_classes needs to be defined here
            num_cls=len(dataset_params['classes']),
            normalize_targets=True,
            post_prediction_callback=PPYoloEPostPredictionCallback(
                score_threshold=0.01,
                nms_top_k=1000,
                max_predictions=300,
                nms_threshold=0.7
            )
        )
    ],
    "metric_to_watch": 'mAP@0.50:0.95',
    "max_epochs": 10
}`

`trainer.train(model=model, 
              training_params=train_params, 
              train_loader=train_data, 
              valid_loader=val_data)`

`trainer.test(model=best_model,
            test_loader=test_data,
            test_metrics_list=DetectionMetrics_050_095(score_thres=0.8, 
                                                   top_k_predictions=300, 
                                                   num_cls=len(dataset_params['classes']), 
                                                   normalize_targets=True, 
                                                   post_prediction_callback=PPYoloEPostPredictionCallback(score_threshold=0.01, 
                                                                                                          nms_top_k=1000, 
                                                                                                          max_predictions=300,                                                                              
                                                                                                          nms_threshold=0.7)
                                                  ))`

![image](https://github.com/Deci-AI/super-gradients/assets/65900166/08c9c253-16b2-4f38-8613-672e016e1873)

**Comment by BloodAxe** [comment]
The `score_thres=0.8` that you have set in the metric has nothing to do with the model predictions.

A `score_thres` in the metric class tells the metric to use only predictions with threshold above given value. In other words the metric will discard predictions with confidence score less than 0.8 in your example. But this does not prevent model from outputting such predictions in principle at predict().

I'm not sure what were your expectations but I don't see a problem here. You can also set a minimum confidence for predict() method as well to filter out low-confidence predictions if you don't need them.
